snapcraft/internal/lxd.py

```diff
@@ -145,60 +145,44 @@ def _setup_project(self):
         self._container_run(['tar', 'xvf', os.path.basename(tar_filename)])
 
     def _inject_snapcraft(self):
-        if common.is_snap():
+        if is_snap():
             # Because of https://bugs.launchpad.net/snappy/+bug/1628289
             self._container_run(['apt-get', 'install', 'squashfuse', '-y'])
 
             # Push core snap into container
             self._inject_snap('core')
-            self._inject_snap('snapcraft', rev=os.environ.get('SNAP_REVISION'))
+            self._inject_snap('snapcraft')
         else:
             self._container_run(['apt-get', 'install', 'snapcraft', '-y'])
 
-    def _inject_snap(self, name, *, rev=None):
+    def _inject_snap(self, name):
         session = requests_unixsocket.Session()
         snapd_socket = '/run/snapd.socket'.replace('/', '%2F')
+        # Cf. https://github.com/snapcore/snapd/wiki/REST-API#get-v2snapsname
         api = 'http+unix://{}/v2/snaps/{}'.format(snapd_socket, name)
         try:
             json = session.request('GET', api).json()
         except Exception as e:
-            raise RuntimeError('Error connecting to {}'.format(api)) from e
+            raise SnapcraftEnvironmentError(
+                'Error connecting to {}'.format(api)) from e
         if not json['status'] == 'OK':
-            raise RuntimeError('Error querying {} snap: {}'.format(name, json))
-        id = json['result']['id']
+            raise SnapcraftEnvironmentError(
+                'Error querying {r!} snap: {}'.format(name, json))
         classic = json['result']['confinement'] == 'classic'
-        if not rev:
-            rev = json['result']['revision']
-
-        self._inject_assertions('{}_{}.assert'.format(name, rev), [
-            ['account-key', 'public-key-sha3-384={}'.format(_STORE_KEY)],
-            ['snap-declaration', 'snap-name={}'.format(name)],
-            ['snap-revision', 'snap-revision={}'.format(rev),
-             'snap-id={}'.format(id)],
-        ])
+        rev = json['result']['revision']
 
         installed = os.path.join(os.path.sep, 'var', 'lib', 'snapd', 'snaps',
                                  '{}_{}.snap'.format(name, rev))
         filename = os.path.basename(installed)
         # Copy file to ensure LXD snap can access it
-        shutil.copyfile(installed, filename)
+        copyfile(installed, filename)
```

Comment:
Since this copies it to cwd, should we remove it after we've pushed it into the container?